src/notion/cache.py

```diff
@@ -1,20 +1,50 @@
+from datetime import datetime
+from datetime import timedelta
 from typing import Callable
 
-from django.core.cache import cache
+from django.utils import timezone
 
 from app.current_user import get_current_user
 from notion.client import NotionClient
+from notion.models import NotionCacheEntry
 from notion.page import NotionPage
 
 TIMEOUT = 60 * 60 * 24 * 14  # 14 days
 
 
-def cache_key(page_id: str) -> str:
-    return f"notion-page-cache--{page_id}"
+class NotionCache:
+    def set(self, cache_key: str, content: NotionPage | Callable[[], NotionPage]) -> NotionPage:
+        expires_datetime = self.get_expires_time()
+        content = self.get_content_as_notion_page(content)
+        content_to_save = content.to_json()
+        NotionCacheEntry.objects.update_or_create(cache_key=cache_key, defaults=dict(content=content_to_save, expires=expires_datetime))
+        return content
 
+    def get(self, cache_key: str) -> NotionPage | None:
+        cache_entry = self._get(cache_key)
+        if cache_entry:
+            return NotionPage.from_json(cache_entry.content)
 
-def invalidate_cache(page_id: str) -> None:
-    cache.delete(cache_key(page_id))
+    def get_or_set(self, cache_key: str, content: NotionPage | Callable[[], NotionPage]) -> NotionPage:
+        cache_entry = self._get(cache_key)
+        if cache_entry:
+            return NotionPage.from_json(cache_entry.content)
+        return self.set(cache_key, content)
+
+    @staticmethod
+    def _get(cache_key: str) -> NotionCacheEntry | None:
+        return NotionCacheEntry.objects.not_expired().filter(cache_key=cache_key).first()
+
+    @staticmethod
+    def get_content_as_notion_page(content: NotionPage | Callable[[], NotionPage]) -> NotionPage:
+        return content() if callable(content) else content
+
+    @staticmethod
+    def get_expires_time(timeout: int = TIMEOUT) -> datetime:
```

Comment:
Не понимаю, зачем здесь передавать переменную на вход. Сейчас приложение это никак не использует
```suggestion
    def get_expires_time() -> datetime:
```